tests/components/youtube/test_config_flow.py

```diff
@@ -72,122 +63,83 @@ async def test_full_flow(
 
 
-@pytest.mark.parametrize(
-    ("fixture", "abort_reason", "placeholders", "calls", "access_token"),
-    [
-        ("get_channel", "reauth_successful", None, 1, "updated-access-token"),
-        (
-            "get_channel_2",
-            "wrong_account",
-            {"title": "Linus Tech Tips"},
-            0,
-            "mock-access-token",
-        ),
-    ],
-)
-async def test_reauth(
+async def test_flow_http_error(
     hass: HomeAssistant,
-    hass_client_no_auth,
-    aioclient_mock: AiohttpClientMocker,
-    current_request_with_host,
-    config_entry: MockConfigEntry,
-    fixture: str,
-    abort_reason: str,
-    placeholders: dict[str, str],
-    calls: int,
-    access_token: str,
+    hass_client_no_auth: ClientSessionGenerator,
+    current_request_with_host: None,
 ) -> None:
-    """Test the re-authentication case updates the correct config entry.
-
-    Make sure we abort if the user selects the
-    wrong account on the consent screen.
-    """
-    config_entry.add_to_hass(hass)
-
-    config_entry.async_start_reauth(hass)
-    await hass.async_block_till_done()
-
-    flows = hass.config_entries.flow.async_progress()
-    assert len(flows) == 1
-    result = flows[0]
-    assert result["step_id"] == "reauth_confirm"
-
-    result = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+    """Check full flow."""
+    result = await hass.config_entries.flow.async_init(
+        "youtube", context={"source": config_entries.SOURCE_USER}
+    )
     state = config_entry_oauth2_flow._encode_jwt(
         hass,
         {
             "flow_id": result["flow_id"],
             "redirect_uri": "https://example.com/auth/external/callback",
         },
     )
+
     assert result["url"] == (
         f"{GOOGLE_AUTH_URI}?response_type=code&client_id={CLIENT_ID}"
         "&redirect_uri=https://example.com/auth/external/callback"
         f"&state={state}&scope={'+'.join(SCOPES)}"
         "&access_type=offline&prompt=consent"
     )
+
     client = await hass_client_no_auth()
     resp = await client.get(f"/auth/external/callback?code=abcd&state={state}")
     assert resp.status == 200
     assert resp.headers["content-type"] == "text/html; charset=utf-8"
 
-    aioclient_mock.clear_requests()
-    aioclient_mock.post(
-        GOOGLE_TOKEN_URI,
-        json={
-            "refresh_token": "mock-refresh-token",
-            "access_token": "updated-access-token",
-            "type": "Bearer",
-            "expires_in": 60,
-        },
-    )
-
     with patch(
-        "homeassistant.components.youtube.async_setup_entry", return_value=True
-    ) as mock_setup, patch(
-        "httplib2.Http.request",
-        return_value=(
-            Response({}),
-            bytes(load_fixture(f"youtube/{fixture}.json"), encoding="UTF-8"),
+        "homeassistant.components.youtube.config_flow.build",
+        side_effect=HttpError(
+            Response(
+                {
```

Comment:
Optional: I'd suggest extracting constants for these large dicts.

Reply:
I don't know, maybe a constant somewhere? I don't have a strong opinion about this so feel free to do later or not.